Formula/shared-mime-info.rb

```diff
@@ -42,10 +42,11 @@ def install
   end
 
   def post_install
-    ln_sf HOMEBREW_PREFIX/"share/mime", share/"mime"
-    (HOMEBREW_PREFIX/"share/mime/packages").mkpath
-    cp (pkgshare/"packages").children, HOMEBREW_PREFIX/"share/mime/packages"
-    system bin/"update-mime-database", HOMEBREW_PREFIX/"share/mime"
+    global_mime = HOMEBREW_PREFIX/"share/mime"
+    ln_sf global_mime, share/"mime" unless File.exist?(share/"mime")
+    (global_mime/"packages").mkpath
+    cp (pkgshare/"packages").children, global_mime/"packages"
```

Comment:
This should also maybe be in the `if` to avoid copying stuff that already exists.

Reply:
We want `shared-mime-info` to be able to ship its own files in that directory, but at the same time make it so other files are upgrade-proof. So we “hide” the files in `pkgshare` in the first place, then copy them back. Symlink would probably not survive `shared-mime-info` upgrade, I guess.

Reply:
To me the main issue here is that if `shared-mime-info` removes files in `packages` then we will never remove them and, because we copied and not symlinked them, will have no way to clean them up. They will also not be removed on an `uninstall`. 

Where do other packages put files? `share/mime` or `share/mime/packages` or both? It seems the best fit might be ensuring that whichever needs to be are directories (and the directories not symlinked) in Homebrew/brew and the files within the directories are symlinked across.